src/libs/schedule.js

```diff
@@ -327,7 +327,7 @@ function getSchedule(db, user, date, callback) {
 			asyncLib.parallel({
 				// Get Portal calendar feed
 				portal: asyncCallback => {
-					portal.getCal(db, user, (err, hasURL, cal) => {
+					portal.getFromCal(db, user, (err, hasURL, cal) => {
```

Comment:
In `schedule.js`, we'd want to use the cache instead of querying the feed, because that takes time and that's the whole point of caching in the first place. Also, if the cache is empty, we'd want to add the user to the Portal queue if they have a URL. If the feed is fetched immediately, we'd want to use the newly cached schedule with the algorithm. If portal doesn't immediately return a non-empty feed, we should fall back to using the default algorithm as if the user hasn't entered their URL yet.